contracts/libraries/PostLib.sol

```diff
@@ -800,7 +801,85 @@ library PostLib  {
                 usersRating[1].rating *= -1;  
             }
         }
-        UserLib.updateUsersRating(userContext, usersRating); 
+        UserLib.updateUsersRating(userContext, usersRating);
+    }
+
+    function changePostType(
+        PostCollection storage self,
+        UserLib.UserContext storage userContext,
+        address userAddr,
+        uint256 postId,
+        PostType newPostType
+    ) public {
+        PostContainer storage postContainer = getPostContainer(self, postId);
+        UserLib.User storage user = UserLib.getUserByAddress(userContext.users, userAddr);
+        SecurityLib.checkRatingAndEnergy(
+            userContext.roles,
+            user,
+            userAddr,
+            userAddr,
+            postContainer.info.communityId,
+            SecurityLib.Action.changePostType
+        );
+        require(newPostType != postContainer.info.postType, "This post type is already set.");
+        
+        VoteLib.StructRating memory oldTypeRating = getTypesRating(postContainer.info.postType);
+        VoteLib.StructRating memory newTypeRating = getTypesRating(newPostType);
+
+        uint8 positive;
+        uint8 negative;
+        for (uint32 i; i < postContainer.votedUsers.length; i++) {
```

Comment:
Do we really need this loop?
Can't we just check `rating` of the post?

`int32 changeUserRating += (newTypeRating.upvotedPost - oldTypeRating.upvotedPost) * postContainer.rating`